spanner/read.go

```diff
@@ -90,6 +91,14 @@ func streamWithReplaceSessionFunc(
 	}
 }
 
+// Iterator is an interface for iterating over Rows.
+type Iterator interface {
```

Comment:
Do we need this interface? (Or put another way: What are the benefits of adding this interface?)

The name also seems very generic.

Reply:
Updated the interface to package private so that can't be used by customers, added this to ease mock and unit tests